pkg/cmd/tknpac/webhook/github.go

```diff
@@ -0,0 +1,161 @@
+package webhook
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+
+	"github.com/AlecAivazis/survey/v2"
+	"github.com/google/go-github/v43/github"
+	"github.com/openshift-pipelines/pipelines-as-code/pkg/cli/prompt"
+	"github.com/openshift-pipelines/pipelines-as-code/pkg/formatting"
+	"golang.org/x/oauth2"
+)
+
+const defaultPublicGithub = "https://github.com"
+
+type gitHubWebhookConfig struct {
+	ControllerURL       string
+	RepoOwner           string
+	RepoName            string
+	WebhookSecret       string
+	PersonalAccessToken string
+	APIURL              string
+}
+
+func (w Webhook) githubWebhook(ctx context.Context) (string, error) {
+	msg := "Would you like me to configure GitHub Webhook for your repository? "
+	var configureWebhook bool
+	if err := prompt.SurveyAskOne(&survey.Confirm{Message: msg, Default: true}, &configureWebhook); err != nil {
+		return "", err
+	}
+	if !configureWebhook {
+		return "", nil
+	}
+
+	ghWebhook, err := askGHWebhookConfig(w.RepositoryURL, w.ControllerURL)
+	if err != nil {
+		return "", err
+	}
+	ghWebhook.APIURL = w.ProviderAPIURL
+
+	return ghWebhook.ControllerURL, ghWebhook.create(ctx)
+}
+
+func askGHWebhookConfig(repoURL, controllerURL string) (*gitHubWebhookConfig, error) {
+	gh := &gitHubWebhookConfig{}
+
+	var repo, defaultRepo string
+	if repoURL != "" {
+		if repo, _ := formatting.GetRepoOwnerFromGHURL(repoURL); repo != "" {
+			defaultRepo = repo
+		}
+	}
+	if defaultRepo != "" {
+		msg := fmt.Sprintf("Please enter the repository which needs to be configured (default: %s):", defaultRepo)
+		if err := prompt.SurveyAskOne(&survey.Input{Message: msg}, &repo); err != nil {
+			return nil, err
+		}
+	} else {
+		msg := "Please enter the repository which needs to be configured (eg. repo-owner/repo-name) : "
+		if err := prompt.SurveyAskOne(&survey.Input{Message: msg}, &repo,
+			survey.WithValidator(survey.Required)); err != nil {
+			return nil, err
+		}
+	}
+
+	if repo == "" {
+		repo = defaultRepo
+	}
+	repoArr := strings.Split(repo, "/")
+	if len(repoArr) != 2 {
+		return nil, fmt.Errorf("invalid repository, needs to be in format 'org-name/repo-name'")
+	}
+
+	gh.RepoOwner = repoArr[0]
+	gh.RepoName = repoArr[1]
+
+	if controllerURL == "" {
+		if err := prompt.SurveyAskOne(&survey.Input{
+			Message: "Please enter your controller public route URL: ",
+		}, &gh.ControllerURL, survey.WithValidator(survey.Required)); err != nil {
+			return nil, err
+		}
+	}
+
+	if err := prompt.SurveyAskOne(&survey.Input{
+		Message: "Please enter secret for configuring with webhook for payload validation: ",
+	}, &gh.WebhookSecret, survey.WithValidator(survey.Required)); err != nil {
+		return nil, err
+	}
+
+	// nolint:forbidigo
+	fmt.Println("🔑 Next, you need to create a GitHub access token with scopes `public_repo` & `admin:repo_hook`")
```

Comment:
the phrasing maybe can be: You need to create a GitHub personal token with scopes  

```
public_repo` & `admin:repo_hook`\nClick here to generate one https://github.com/settings/tokens/new and see https://is.gd/BMgLH5 for documentation 
```